doc/source/templates/tests/batch_inference.ipynb

```diff
@@ -0,0 +1,371 @@
+    "            result = self.model(input_data)\n",
+    "        return {\"predictions\": result.cpu().numpy()}\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fda0c298",
+   "metadata": {},
+   "source": [
+    "Now, perform batch prediction using Ray Data! Ray Data will perform model inference using `NUM_WORKERS` copies of the `PredictCallable` class you defined."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "331e21e4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictions = ds.map_batches(\n",
+    "    PredictCallable,\n",
+    "    batch_size=128,\n",
+    "    compute=ray.data.ActorPoolStrategy(\n",
+    "        # Fix the number of batch inference workers to a specified value.\n",
+    "        size=NUM_WORKERS,\n",
+    "    ),\n",
+    "    num_gpus=NUM_GPUS_PER_WORKER,\n",
+    "    batch_format=\"numpy\",\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "23e77ada",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "preds = predictions.materialize()\n",
+    "preds.schema()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d606556",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "preds.take(1)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ceddd984",
+   "metadata": {},
+   "source": [
+    "```{tip}\n",
+    "Play around with the `min_size` and `max_size` parameters to enable autoscaling!\n",
+    "For example, try commenting out `max_size`: this will autoscale up to an infinite number of workers, if you have free resources in the cluster.\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "90ec67e8",
+   "metadata": {},
+   "source": [
+    "Shard the predictions into a few partitions, and save each partition to a file!\n",
+    "\n",
+    "```{note}\n",
+    "This currently saves to the local filesystem under `/tmp/predictions`, but you could also save to a cloud bucket (e.g., `s3://predictions-bucket`).\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c1887e34",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "num_shards = 3\n",
+    "\n",
+    "with tempfile.TemporaryDirectory() as temp_dir:\n",
+    "    predictions.repartition(num_shards).write_parquet(f\"local://{temp_dir}\")\n",
+    "    print(f\"Predictions saved to `{temp_dir}`!\")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1e88a268",
+   "metadata": {
+    "tags": [
+     "test"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import os\n",
+    "\n",
+    "release_test_out = os.environ.get(\"TEST_OUTPUT_JSON\", \"/tmp/release_test_out.json\")\n",
+    "\n",
+    "elapsed = time.monotonic() - start_time\n",
+    "with open(release_test_out, \"wt\") as f:\n",
+    "    json.dump({\"total_runtime\": elapsed}, f)\n"
+   ]
```

Comment:
Should we expect `remove-cell` to be included in this PR?

Reply:
Oh yeah, I can add it in. It's not really used right now, but will be needed in the future once de-dupe the notebooks.

Reply:
Ah okay this question was because `remove-cell` was mentioned in the contributor guide.